sdv/data_processing/data_processor.py

```diff
@@ -720,6 +717,13 @@ def reverse_transform(self, data, reset_keys=False):
             generated_keys = self.generate_keys(num_rows, reset_keys)
             sampled_columns.extend(self._keys)
 
+        for constraint in reversed(self._constraints_to_reverse):
+            reversed_data = constraint.reverse_transform(reversed_data)
```

Comment:
would it work if you just did sampled_columns = constraint.reverse_transform(sampled_columns)?

Reply:
Sorry, I don't get what you mean, you can't apply `reverse_transform` to `sampled_columns`, since that's a list.

Reply:
Yeah you're right, I got confused. Should we just set `sampled_columns` to be `reversed_data.columns` on line 725 instead of extending. Otherwise that list might contain columns that were created by constraints that we don't want

Reply:
That was actually the first thing I tried. It doesn't work because `sampled_columns` has the keys, which `reversed_data.columns` doesn't have.

Your concern is actually not a problem because of line 731, it removes any columns not seen in the metadata.I originally had an extra two lines to remove these extra columns directly, for clarity, but I decided to remove them since they were unnecessary.

Reply:
That makes sense. Can you edit the comments before line 731 to add that extra columns will also be dropped?